src/logging.cpp

```diff
@@ -21,7 +21,11 @@ const char * const DEFAULT_DEBUGLOGFILE = "debug.log";
  * This method of initialization was originally introduced in
  * ee3374234c60aba2cc4c5cd5cac1c0aefc2d817c.
  */
-BCLog::Logger* const g_logger = new BCLog::Logger();
+BCLog::Logger& LogInstance()
+{
+    static BCLog::Logger* g_logger{new BCLog::Logger()};
```

Comment:
Would add comment here that this intentionally never calls logger destructor. Would also add a comment in the logger class saying that if a destructor is defined there it will never be called, and that any members added to the logger class will not have their destructors called either.

Reply:
The comment above says that, I think. Though I added a line clarifying that it must be a trivial destructor.